src/webviews/settings/index.ts

```diff
@@ -171,34 +163,20 @@ export class SettingsUI {
             }
           }
 
-          if (server) {
-            if (connectionSettings && existingConfigIndex >= 0) {
-              config = {
-                ...config,
-                ...data,
-              };
-
-              connectionSettings[existingConfigIndex] = config;
-              await GlobalConfiguration.set(`connectionSettings`, connectionSettings);
-            }
-          } else {
-            if (connection) {
-              if (restartFields.some(item => data[item] !== config[item])) {
-                restart = true;
-              }
-
-              Object.assign(config, data);
-              await ConnectionConfiguration.update(config);
-            }
+          if (restartFields.some(item => data[item] !== config[item])) {
+            restart = true;
           }
 
-          if (restart) {
-            vscode.window.showInformationMessage(`Some settings require a restart to take effect. Reload workspace now?`, `Reload`, `No`)
-              .then(async (value) => {
-                if (value === `Reload`) {
-                  await vscode.commands.executeCommand(`workbench.action.reloadWindow`);
-                }
-              });
+          Object.assign(config, data);
+          await ConnectionConfiguration.update(config);
```

Comment:
We should use `instance.setConfig` here. It will call `ConnectionConfiguration.update` and also update the current connection's config if applicable. Otherwise, `instance.getConfig` will return a stale config.
```suggestion
          await instance.setConfig(config);
```